src/driver.h

```diff
@@ -126,8 +126,13 @@
 
 #define TEGRA_ARRAY_SIZE(x) (sizeof(x) / sizeof(*x))
 
-#define __TEGRA_ALIGN_MASK(x, mask) (((x) + (mask)) & ~(mask))
-#define TEGRA_ALIGN(x, a)           __TEGRA_ALIGN_MASK(x, (typeof(x))(a) - 1)
+#define __TEGRA_ALIGN_MASK(x, y)    ((__typeof__(x))((y)-1))
+#define TEGRA_ROUND_UP(x, y)        ((((x)-1) | __TEGRA_ALIGN_MASK(x, y))+1)
+#define TEGRA_ROUND_DOWN(x, y)      ((x) & ~__TEGRA_ALIGN_MASK(x, y))
+
+#define TEGRA_ALIGN(x, a)           TEGRA_ROUND_UP(x, a)
```

Comment:
Compiler may precompute macro only if `a` is a constant value. I think it is not always constant in our case.

Reply:
Right, because we align with the result of getpagesize()... But, to be a bit correctness-pedantic here, getpagesize() isn't guaranteed to return a power of two, which your version requires for correct behavior. But more realistically, since this code is always running on 32-bit ARM, this is [effectively a constant](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arm/include/asm/page.h).

Anyway, this gets very theoretical very quickly. I just don't like the complexity of the code, because it's rather tricky to decipher IMO. But I see your point, though, and perhaps it's the right speed vs theoretical problem trade-off.